### PR TITLE
Build an EmbedIndex tool package during the build

### PR DESCRIPTION
**Pull request by dagood**
Restores/publishes using the dotnet tool, independent from the "main" build. Ideally it would use existing artifacts from the main build, but it creates its own. The NuGet package is generated normally, with a new nuspec.

@noahfalk @tmat

### CODE REVIEW

**Comment by dagood** [comment]
Ah, well those test failures seem to explain why the main solution doesn't include EmbedIndex--looks like CLI isn't installed. I can download it during the build, unless that sounds unreasonable.

**Comment by tmat** [comment]
Yes, feel free to download it.

File: build/Targets/Imports.targets [hunk 1]
```diff
@@ -7,4 +7,5 @@
   <Import Project="..\Toolset\DefaultImports.targets"/>
   <Import Project="..\Toolset\GenerateAssemblyInfo.targets" />
   <Import Project="..\Toolset\GenerateInternalsVisibleTo.targets" />
+  <Import Project="..\Toolset\DotNetPublish.targets" />
```
Comment (tmat) [inline build/Targets/Imports.targets]: I'd prefer if this Import was directly in the EmbedIndex project rather then global for all projects.

**Comment by dagood** [comment]
Thanks, made those changes and got CI to pass.

**Comment by noahfalk** [comment]
LGTM

**Pull request merged**